networkit/cpp/centrality/ApproxBetweenness.cpp

```diff
@@ -46,76 +45,72 @@ void ApproxBetweenness::run() {
     INFO("estimated diameter: ", vd);
     r = ceil((universalConstant / (epsilon * epsilon)) * (floor(log2(vd - 2)) + 1 - log(delta)));
 
+    std::vector<std::atomic<double>> scores(G.upperNodeIdBound());
+    sssps.resize(omp_get_max_threads());
+#pragma omp parallel
+    {
+        omp_index i = omp_get_thread_num();
+        if (G.isWeighted())
+            sssps[i] = std::unique_ptr<SSSP>(new Dijkstra(G, 0, true, false));
+        else
+            sssps[i] = std::unique_ptr<SSSP>(new BFS(G, 0, true, false));
+    }
+
     INFO("taking ", r, " path samples");
-    // parallelization:
-    count maxThreads = omp_get_max_threads();
-    DEBUG("max threads: ", maxThreads);
-    std::vector<std::vector<double> > scorePerThread(maxThreads, std::vector<double>(G.upperNodeIdBound()));
-    DEBUG("score per thread size: ", scorePerThread.size());
     handler.assureRunning();
     #pragma omp parallel for
     for (omp_index i = 1; i <= static_cast<omp_index>(r); i++) {
-        count thread = omp_get_thread_num();
         DEBUG("sample ", i);
-        // if (i >= 1000) throw std::runtime_error("too many iterations");
-        // DEBUG
         // sample random node pair
         node u, v;
         u = GraphTools::randomNode(G);
         do {
             v = GraphTools::randomNode(G);
         } while (v == u);
 
-        // runs faster for unweighted graphs
-        std::unique_ptr<SSSP> sssp;
-        if (G.isWeighted()) {
-            sssp.reset(new Dijkstra(G, u, true, false, v));
-        } else {
-            sssp.reset(new BFS(G, u, true, false, v));
-        }
+        auto &sssp = *(sssps[omp_get_thread_num()].get());
```

Comment:
In my opinion, it would be more elegant to initialize the SSSP instance in an outer `omp parallel` section, in which then the `omp for` is nested. That way, the global vector could be avoided.

Reply:
That's a good idea, I've implemented it.